src/victory-primitives/curve.js

```diff
@@ -56,24 +56,43 @@ export default class Curve extends React.Component {
   }
 
   getDataSegments(data) {
-    let segmentStartIndex = 0;
-    const segments = data.reduce((memo, datum, index) => {
+    return this.splitArray(data, (datum) => {
       const yDatum = datum.y1 !== undefined ? datum._y1 : datum._y;
-      if (yDatum === null || typeof yDatum === "undefined") {
-        memo = memo.concat([data.slice(segmentStartIndex, index)]);
+
+      return yDatum === null || typeof yDatum === "undefined";
+    }).filter((segment) => segment.length > 1);
+  }
+
+  /**
+   * Split array into subarrays using a delimiter function. Items qualifying as
+   * delimiters are excluded from the subarrays. Functions similarly to String.split
+   *
+   * Example:
+   * const array = [1, 2, 3, "omit", 4, 5, "omit", 6]
+   * splitArray(array, (item) => item === "omit");
+   * => [[1, 2, 3], [4, 5], [6]]
+   *
+   * @param {Array}    array        An array of items
+   * @param {Function} delimiterFn  A function indicating values to be used as delimiters
+   * @returns {Object}              Array of subarrays
+   */
+  splitArray(array, delimiterFn) {
```

Comment:
Looks good. This can be extracted into `util/collection.js` after your other PR goes in.